cmd/ecloud/ecloud_networkruleport_test.go

```diff
@@ -0,0 +1,473 @@
+package ecloud
+
+import (
+	"errors"
+	"testing"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/ukfast/cli/internal/pkg/clierrors"
+	"github.com/ukfast/cli/test/mocks"
+	"github.com/ukfast/cli/test/test_output"
+	"github.com/ukfast/sdk-go/pkg/service/ecloud"
+)
+
+func Test_ecloudNetworkRulePortList(t *testing.T) {
+	t.Run("DefaultRetrieve", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		service := mocks.NewMockECloudService(mockCtrl)
+
+		service.EXPECT().GetNetworkRulePorts(gomock.Any()).Return([]ecloud.NetworkRulePort{}, nil).Times(1)
+
+		ecloudNetworkRulePortList(service, &cobra.Command{}, []string{})
+	})
+
+	t.Run("MalformedFlag_ReturnsError", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		service := mocks.NewMockECloudService(mockCtrl)
+		cmd := &cobra.Command{}
+		cmd.Flags().StringArray("filter", []string{"invalidfilter"}, "")
+
+		err := ecloudNetworkRulePortList(service, cmd, []string{})
+
+		assert.IsType(t, &clierrors.ErrInvalidFlagValue{}, err)
+	})
+
+	t.Run("GetNetworkRulePortsError_ReturnsError", func(t *testing.T) {
+
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		service := mocks.NewMockECloudService(mockCtrl)
+
+		service.EXPECT().GetNetworkRulePorts(gomock.Any()).Return([]ecloud.NetworkRulePort{}, errors.New("test error")).Times(1)
+
+		err := ecloudNetworkRulePortList(service, &cobra.Command{}, []string{})
+
+		assert.Equal(t, "Error retrieving network rule ports: test error", err.Error())
+	})
+}
+
+func Test_ecloudNetworkRulePortShowCmd_Args(t *testing.T) {
+	t.Run("ValidArgs_NoError", func(t *testing.T) {
+		err := ecloudNetworkRulePortShowCmd(nil).Args(nil, []string{"fwrp-abcdef12"})
+
+		assert.Nil(t, err)
+	})
+
+	t.Run("InvalidArgs_Error", func(t *testing.T) {
+		err := ecloudNetworkRulePortShowCmd(nil).Args(nil, []string{})
+
+		assert.NotNil(t, err)
+		assert.Equal(t, "Missing network rule port", err.Error())
+	})
+}
+
+func Test_ecloudNetworkRulePortShow(t *testing.T) {
+	t.Run("SingleNetworkRulePort", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		service := mocks.NewMockECloudService(mockCtrl)
+
+		service.EXPECT().GetNetworkRulePort("fwrp-abcdef12").Return(ecloud.NetworkRulePort{}, nil).Times(1)
+
+		ecloudNetworkRulePortShow(service, &cobra.Command{}, []string{"fwrp-abcdef12"})
+	})
+
+	t.Run("MultipleNetworkRulePorts", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		service := mocks.NewMockECloudService(mockCtrl)
+
+		gomock.InOrder(
+			service.EXPECT().GetNetworkRulePort("fwrp-abcdef12").Return(ecloud.NetworkRulePort{}, nil),
```

Comment:
is `fwrp-` the right ID prefix here, or should it be something like `nrp-`?

Reply:
Ahh good spot, updated!